### PR TITLE
Improve Mastodon support

### LINKED ISSUE #92: When using more the one account all interactions are redirected to main account (ignore logged-in instances instead)
## Bug description
I have one account registered with the add on but also use another account on another server. When the add-on is activated, tryting to reply to a post, to boost it or favour leads to an error. When I de-activate the plugin, all works fine. Somebody else noted similar problems and these were resolved when she deactivated the plugin. 

## Steps to reproduce

1. install the plugin, an register a mastodon account
2. open another account on another browser in Firefox
3. for any post click on reply, boot, or favour

### Actual behavior
leads to error

### Expected behavior
do what the buttons are supposed to to

## System

Operating system and version: 
Browser and version: 107.0.1 
Add-on version: 2.0

### PR DESCRIPTION
**Pull request by austinhuang0131**
The commits should be self-explanatory.

closes #92

### CODE REVIEW

File: src/content_script/mastodonInject.js [hunk 1]
```diff
@@ -84,6 +84,21 @@ function waitForElement(selector, multiple = false, timeoutDuration = 200000) {
     });
 }
 
+/**
+ * Checks whether the user has logged into the instance.
+ * 
+ * @returns {boolean}
+ */
+async function checkLoggedIn() {
```
Comment (rugk) [inline src/content_script/mastodonInject.js:92]: `isLoggedIn` is maybe the better name?

File: src/content_script/mastodonInject.js [hunk 2]
```diff
@@ -137,8 +152,12 @@ async function injectInteractionButtons() {
...
  */
 function initInjections() {
-    injectFollowButton().catch(console.error);
-    injectInteractionButtons().catch(console.error);
+    checkLoggedIn().then((r) => {
+        if (!r) {
+            injectFollowButton().catch(console.error);
+            injectInteractionButtons().catch(console.error);
+        }
+    });
```
Comment (rugk) [inline src/content_script/mastodonInject.js:160]:
```suggestion
    checkLoggedIn().then((isLoggedIn) => {
        if (isLoggedIn) {
            return;
        }
        
        injectFollowButton().catch(console.error);
        injectInteractionButtons().catch(console.error);
    });
```

early return + more clean code IMHO.

Or please actually even better with await:
```suggestion
    const isLoggedIn = await checkLoggedIn();
    if (isLoggedIn) {
        return;
    }
        
        
    injectFollowButton().catch(console.error);
    injectInteractionButtons().catch(console.error);
```

+ make the function async

**Review by rugk — CHANGES_REQUESTED** [review]
Some minor changes.

File: src/content_script/mastodonInject.js (hunk 2 above)
Comment (austinhuang0131) [reply src/content_script/mastodonInject.js]: Making it async/await throws a ReferenceError (accessing before initialization) so the first choice is taken.

Comment (rugk) [reply src/content_script/mastodonInject.js]: > Making it async/await throws a ReferenceError (accessing before initialization)

Uff, why? Actually that should just be a different syntax for the same result.

**Review by rugk — APPROVED** [review]

File: src/content_script/mastodonInject.js (hunk 2 above)
Comment (austinhuang0131) [reply src/content_script/mastodonInject.js]:
```js
async function initInjections() {
    const isLoggedIn = await isLoggedIn();
    if (isLoggedIn) {
        return;
    }
    injectFollowButton().catch(console.error);
    injectInteractionButtons().catch(console.error);
}
```

Screenshot 2023-03-19 at 10 57 35

Comment (rugk) [reply src/content_script/mastodonInject.js]: crazy…

**Pull request merged**